### PR TITLE
Shield Database

### PR DESCRIPTION
**Pull request by panleone**
Add a standard way to export and import shieldData

### CODE REVIEW

**Comment by panleone** [comment]
ready for review! I'm updating MPW so testing become easier

File: js/pivx_shielding.js [hunk 1]
```diff
@@ -80,14 +82,23 @@ export class PIVXShielding {
...
-      blockHeight,
-      isTestNet
-    );
-    pivxShielding.lastProcessedBlock = effectiveHeight;
-    pivxShielding.commitmentTree = commitmentTree;
+    let readFromData = false;
+    if (data) {
+      let shieldDB = JSON.parse(data);
+      if (await pivxShielding.load(shieldDB)) {
+        console.log("Successfully read database!");
```
Comment (Duddino) [inline js/pivx_shielding.js:89]:
```suggestion
```

File: js/pivx_shielding.js [hunk 2]
```diff
@@ -353,3 +399,28 @@ export class UTXO {
...
+   * Add a transparent UTXO, along with its private key
+   * @param {Object} o - Options
+   * @param {String} o.sanityAddress - A sanity sapling shield address
+   * @param {Number} o.lastProcessedBlock - Last processed block in blockchain
+   * @param {String} o.commitmentTree - Hex encoded commitment tree
+   * @param {Uint8Array} o.diversifierIndex - Diversifier index of the last generated address
+   * @param {[Note, String][]} o.unspentNotes - Array of notes, corresponding witness
+   */
+  constructor({
+    sanityAddress,
```
Comment (Duddino) [inline js/pivx_shielding.js:414]: Sanity address doesn't mean anything to me, maybe call it default address?

File: js/pivx_shielding.js [hunk 3]
```diff
@@ -353,3 +399,28 @@ export class UTXO {
     this.script = script;
   }
 }
+
+export class ShieldDB {
```
Comment (Duddino) [inline js/pivx_shielding.js:403]: I don't really like this name, this is not a database, it shouldn't be called DB

**Review by Duddino — CHANGES_REQUESTED** [review]
Good work, but as always I have my nitpicks :smiley:

File: js/pivx_shielding.js [hunk 4]
```diff
@@ -148,7 +158,44 @@ export class PIVXShielding {
...
+      })
+    );
+  }
 
+  /**
+   * Load shieldWorker from a shieldData
+   * @param {ShieldData} shieldData - shield data
+   */
+  async load(shieldData) {
+    const { defAddress, _ } = await this.callWorker(
```
Comment (Duddino) [inline js/pivx_shielding.js:185]:
```suggestion
    const { address, _ } = await this.callWorker(
```

**Comment by panleone** [comment]
suggestion fixed

**Review by Duddino — APPROVED** [review]
tACK

**Pull request merged**